### PR TITLE
Use K8sNameDescriptionField in connection type page

### PR DESCRIPTION
**Pull request by emilys314**
Closes https://issues.redhat.com/browse/RHOAIENG-13400

## Description

Use the new `K8sNameDescriptionField` component. This prevents the user from entering a too long of a name which causes a resource name error when saving. 

image
image
image
image

## How Has This Been Tested?

Tested locally by running through the different connection type operations. 
For the bug referenced in the jira issue, paste a lot of text https://www.lipsum.com/feed/html and click save. It will now truncate the resource name

## Test Impact

No updates, existing tests cover the previous functionality. (and the long name is a server side error)

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/k8s/utils.ts [hunk 1]
```diff
@@ -7,7 +7,8 @@ export const PreInstalledName = 'Pre-installed';
 export const ownedByDSC = (resource: K8sResourceCommon): boolean =>
   !!resource.metadata?.ownerReferences?.find((owner) => owner.kind === 'DataScienceCluster');
 
-export const isK8sDSGResource = (x?: K8sResourceCommon): x is K8sDSGResource => !!x?.metadata?.name;
+export const isK8sDSGResource = (x?: K8sResourceCommon): x is K8sDSGResource =>
+  !!x?.metadata && 'name' in x.metadata;
```
Comment (christianvogt) [inline frontend/src/concepts/k8s/utils.ts:11]: Why the change?

**Review by christianvogt — COMMENTED** [review]
I can still submit the form if there are errors with the resource name.

File: frontend/src/concepts/k8s/utils.ts (hunk 1 above)
Comment (emilys314) [reply frontend/src/concepts/k8s/utils.ts:11]: For prefilling the k8sNameDesc fields, it runs the following:
```
export const setupDefaults = ({
  initialData,
  limitNameResourceType,
  safePrefix,
}: UseK8sNameDescriptionDataConfiguration): K8sNameDescriptionFieldData => {
  let initialName = '';
  let initialDescription = '';
  let initialK8sNameValue = '';
  let configuredMaxLength = MAX_RESOURCE_NAME_LENGTH;

  if (isK8sDSGResource(initialData)) {
    initialName = getDisplayNameFromK8sResource(initialData);
    initialDescription = getDescriptionFromK8sResource(initialData);
    initialK8sNameValue = initialData.metadata.name;
  }
  if (limitNameResourceType != null) {
    configuredMaxLength = ROUTE_BASED_NAME_LENGTH;
  }
  ```
  
So specifically for the duplicate scenario, there will be data like the description and display name filled out, but no resource name is defined yet, and it will fail the check since "" is false. This change makes "" still count as true

Comment (emilys314) [reply frontend/src/concepts/k8s/utils.ts]: Okay I reverted this change due to other changes anyways. For duplicate, it will now pass in the resource name, so the type check will always pass.

Passing in the resource name would originally block you from editing it, but i added `editK8sNameValue` to allow edit of the prefilled value
